frontend/lib/uast/post-parse-checks.cpp

```diff
@@ -1290,6 +1295,16 @@ void Visitor::checkLinkageName(const NamedDecl* node) {
   }
 }
 
+void Visitor::checkTupleDeclFormalIntent(const TupleDecl* node) {
+  // 'VAR' might show up in the case of nested tuple decl formals, for example:
```

Comment:
In `Resolver::resolveTupleUnpackDecl` the logic and comment checking for default intent implied to me that we don't need to check for `VAR` intent. I assumed the inner tuples wouldn't be counted as tuple decl formals, so `node->isTupleDeclFormal()` would be false. Is that incorrect? In any case this seems to contradict the previous logic to me.

Reply:
The difference is that in ``Resolver`` we don't "visit" nested ``TupleDecl`` uAST nodes with ``resolveTupleDecl``, and in post-parse-checks we do visit those nodes with ``checkTupleDeclFormalIntent``